pkg/apis/apis/v1alpha1/types.go

```diff
@@ -0,0 +1,232 @@
+}
+
+// ResourceSchemaReference records a resource, identified by (group, version, resource, schema).
+type ResourceSchemaReference struct {
+	metav1.GroupVersionResource `json:",inline"`
+
+	// Schema is the name of the bound ResourceSchema.
+	Schema string `json:"schema,omitempty"`
+	// Hash is
+	Hash string `json:"hash,omitempty"`
+}
+
+// ServiceBindingList is a list of ServiceBinding resources
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ServiceBindingList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []ServiceBinding `json:"items"`
+}
+
+// ServiceExport registers an API and implementation to allow consumption by others.
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories=kcp
+type ServiceExport struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec holds the desired state.
+	// +optional
+	Spec ServiceExportSpec `json:"spec,omitempty"`
+
+	// Status communicates the observed state.
+	// +optional
+	Status ServiceExportStatus `json:"status,omitempty"`
+}
+
+type ServiceExportSpec struct {
+	// ResourceSchemas records which resource schemas are exposed with this ServiceExport.
+	//
+	// +optional
+	// +listType=map
+	// +listMapKey=schema
+	// +patchStrategy=merge
+	// +patchMergeKey=schema
+	ResourceSchemas []ResourceSchemaReference `json:"resourceSchemas,omitempty" patchStrategy:"merge" patchMergeKey:"schema"`
+}
+
+type ServiceExportStatus struct {
+	// ResourceSchemasInUse records which schemas are actually in use (that is, object exist at that schema in some
+	// Workspace that binds this ServiceExport).
+	//
+	// +optional
+	ResourceSchemasInUse []string `json:"resourceSchemasInUse,omitempty"`
+}
+
+// ServiceExportList is a list of ServiceExport resources
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ServiceExportList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []ServiceExport `json:"items"`
+}
+
+// ResourceSchema describes .
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories=kcp
+type ResourceSchema struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec holds the desired state.
+	// +optional
+	Spec ResourceSchemaSpec `json:"spec,omitempty"`
+
+	// Status communicates the observed state.
+	// +optional
+	Status ResourceSchemaStatus `json:"status,omitempty"`
+}
+
+type ResourceSchemaSpec struct {
+	apiextensionsv1.CustomResourceDefinitionSpec `json:",inline"`
```

Comment:
Very unclear as to what we want here and how it is the same or different from CRDSpec...

Reply:
for now it is single version, but other than that nearly all that CRDs offer (no webhooks for now).

Reply:
Just copy-pasta time?